pipelines/processor.go

```diff
@@ -0,0 +1,239 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package pipelines
+
+import (
+	"log"
+	"math"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/DataDog/sketches-go/ddsketch"
+	"github.com/DataDog/sketches-go/ddsketch/mapping"
+	"github.com/DataDog/sketches-go/ddsketch/store"
+	"github.com/golang/protobuf/proto"
+)
+
+const (
+	bucketDuration     = time.Second * 10
+	defaultServiceName = "unnamed-go-service"
+)
+
+var sketchMapping, _ = mapping.NewLogarithmicMapping(0.01)
+
+type statsPoint struct {
+	service        string
+	edge           string
+	hash           uint64
+	parentHash     uint64
+	timestamp      int64
+	pathwayLatency int64
+	edgeLatency    int64
+}
+
+type statsGroup struct {
+	service        string
+	edge           string
+	hash           uint64
+	parentHash     uint64
+	pathwayLatency *ddsketch.DDSketch
+	edgeLatency    *ddsketch.DDSketch
+}
+
+type bucket map[uint64]statsGroup
+
+func (b bucket) export() []groupedStats {
+	stats := make([]groupedStats, 0, len(b))
+	for _, s := range b {
+		pathwayLatency, err := proto.Marshal(s.pathwayLatency.ToProto())
+		if err != nil {
+			log.Printf("ERROR: can't serialize pathway latency. Ignoring: %v", err)
+			continue
+		}
+		edgeLatency, err := proto.Marshal(s.edgeLatency.ToProto())
+		if err != nil {
+			log.Printf("ERROR: can't serialize edge latency. Ignoring: %v", err)
+			continue
+		}
+		stats = append(stats, groupedStats{
+			PathwayLatency: pathwayLatency,
+			EdgeLatency:    edgeLatency,
+			Service:        s.service,
+			Edge:           s.edge,
+			Hash:           s.hash,
+			ParentHash:     s.parentHash,
+		})
+	}
+	return stats
+}
+
+type processorStats struct {
+	payloadsIn      int64
+	flushedPayloads int64
+	flushedBuckets  int64
+	flushErrors     int64
+}
+
+type processor struct {
```

Comment:
This is actually a writer, why not call it `writer`? Or `transport`? "processor" is very generic and can apply to anything that does something.

Reply:
🤔 what do you think about "aggregator"? or "accumulator", or "buffer"?
The goal of this component is to reduce the number of points sent to the agent / Datadog by aggregating them.